tests/unittest_brain_dataclasses.py

```diff
@@ -643,3 +643,39 @@ class A:
     )
     init = next(node.infer())
     assert init._proxied.parent.name == "object"
+
+
+@parametrize_module
+def test_annotated_enclosed_field_call(module: str):
+    """Test inference of dataclass attribute with an annotated field call in another function call"""
+    node = astroid.extract_node(
+        f"""
+    from {module} import dataclass, field
+    from typing import cast
+
+    @dc.dataclass
+    class A:
+        attribute: int = cast(int, field(default_factory=dict))
+
+    """
+    )
+    inferred = node.inferred()
+    assert len(inferred) == 1
+
+
+@parametrize_module
+def test_enclosed_field_call(module: str):
+    """Test inference of dataclass attribute with a non-annotated field call in another function call"""
+    node = astroid.extract_node(
+        f"""
+    from {module} import dataclass, field
+    from typing import cast
+
+    @dc.dataclass
+    class A:
+        attribute = cast(int, field(default_factory=dict))
```

Comment:
Why not combine both tests by adding this as `attribute2` in the first test. Not really necessary to have two separate once for it.

Reply:
Good idea! Going to do this now!